packages/wrangler/src/cloudchamber/build.ts

```diff
@@ -287,3 +300,56 @@ export async function pushCommand(
 		throw new UserError("An unknown error occurred");
 	}
 }
+
+export async function ensureDiskLimits(options: {
+	imageTag: string;
+	pathToDocker: string;
+	account: CompleteAccountCustomer;
+}): Promise<string> {
+	const imageTag = domain + "/" + options.imageTag;
+	const dockerPath = options.pathToDocker ?? "docker";
+	const maxAllowedImageSizeBytes =
+		(options.account.limits.disk_mb_per_deployment ?? 2000) * 1000 * 1000;
+
+	const { size, layers } = await runDockerInspect(imageTag, dockerPath);
+	const adjustedSize = Math.ceil(size * 1.1 + layers * 16 * 1000 * 1000);
+
+	if (maxAllowedImageSizeBytes < adjustedSize) {
+		throw new UserError(
+			`Image too large: needs ${Math.ceil(adjustedSize / 1000 / 1000)} MB, but your account is limited to images with size ${maxAllowedImageSizeBytes / 1000 / 1000} MB. Your account needs more disk size per instance to run this container.`
```

Comment:
Shouldn’t we be checking against the disk size of the application? For example, a user’s account level disk limit could be 20 GB but if the application is configured with a 2 GB disk, then a 2.5 GB image should fail.

Reply:
Now we check the app config and also check account limits during build if the app configuration already exceeds this limit.